ui/src/app/components/runtime-configuration-panel/preset-selector.tsx

```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
+import { Dropdown } from 'primereact/dropdown';
+
+import { Disk } from 'generated/fetch';
+
+import colors from 'app/styles/colors';
+import { runtimePresets } from 'app/utils/runtime-presets';
+import { AnalysisConfig, toAnalysisConfig } from 'app/utils/runtime-utils';
+
+export interface PresetSelectorProps {
+  allowDataproc: boolean;
+  setAnalysisConfig: (analysisConfig: AnalysisConfig) => void;
+  disabled: boolean;
+  gcePersistentDisk: Disk | null | undefined;
+}
+// Select a recommended preset configuration.
+export const PresetSelector = ({
+  allowDataproc,
+  setAnalysisConfig,
+  disabled,
+  gcePersistentDisk,
+}: PresetSelectorProps) => {
+  return (
+    <Dropdown
+      id='runtime-presets-menu'
+      appendTo='self'
+      disabled={disabled}
+      style={{
+        marginTop: '21px',
+        color: colors.primary,
+      }}
+      placeholder='Recommended environments'
+      options={Object.values(runtimePresets)
```

Comment:
this is changed from the original: I converted the lodash/fp code to the equivalent plain JS